aiohttp/http_exceptions.py

```diff
@@ -85,11 +85,10 @@ def __init__(
 
 class InvalidHeader(BadHttpMessage):
     def __init__(self, hdr: Union[bytes, str]) -> None:
-        if isinstance(hdr, bytes):
-            hdr = hdr.decode("utf-8", "surrogateescape")
-        super().__init__(f"Invalid HTTP Header: {hdr}")
-        self.hdr = hdr
-        self.args = (hdr,)
+        hdr_s = hdr.decode(errors="backslashreplace") if isinstance(hdr, bytes) else hdr
+        super().__init__(f"Invalid HTTP header: {hdr_s}")
```

Comment:
My suggestion used a new variable name, so we could use the repr() here, thus resolving the other minor security issue.
```suggestion
        super().__init__(f"Invalid HTTP header: {hdr!r}")
```

Reply:
Oh sorry; I see what you were going for. Fixed now.

Reply:
(I didn't just commit the suggestion because it requires updating tests too)